src/file_auth.ts

```diff
@@ -0,0 +1,48 @@
+import fs = require('fs');
+import https = require('https');
+import request = require('request');
+
+import { Authenticator } from './auth';
+import { User } from './config_types';
+
+export class FileAuth implements Authenticator {
+    private token: string | null = null;
+    private lastRead: Date | null = null;
+
+    public isAuthProvider(user: User): boolean {
+        return user.authProvider && user.authProvider.config && user.authProvider.config.tokenFile;
+    }
+
+    public getToken(user: User): string | null {
```

Comment:
I noticed https://github.com/kubernetes-client/javascript/blob/master/src/auth.ts#L8.  Is it worth rolling this `getToken` implementation into `applyAuthentication` now?